python/src/phylanx.cpp

```diff
@@ -164,12 +170,59 @@ namespace phylanx { namespace bindings
 
         return phylanx::execution_tree::primitive_operand(f.arg_);
     }
-}}
+}
+}
 
 ///////////////////////////////////////////////////////////////////////////////
 void init_hpx_runtime();
 void stop_hpx_runtime();
 
+const char* expression_compiler_help =
+    "compile and evaluate a numerical expression in Phylanx lisp";
+template <typename... Ts>
+typename hpx::util::invoke_result<phylanx::execution_tree::primitive(Ts...),
+    Ts...>::type
+expression_compiler(std::string xexpr, Ts const&... ts)
+{
+    return hpx::threads::run_as_hpx_thread([&]() {
+        using namespace phylanx::execution_tree;
+        auto here = hpx::find_here();
+        try
+        {
+            phylanx::execution_tree::compiler::function_list eval_snippets;
+            auto x = phylanx::execution_tree::compile(
+                xexpr, eval_snippets);    //, env);
+
+            phylanx::execution_tree::primitive_argument_type result;
+            result = x(ts...);
+            int ndx = result.index();
+            if (ndx == 4)
+            {
+                auto node_result =
+                    phylanx::util::get<phylanx::ir::node_data<double>>(result);
+                return primitive{
+                    hpx::local_new<primitives::variable>(node_result)};
+            }
+            else if (ndx == 2)
+            {
+                auto node_result = phylanx::util::get<std::int64_t>(result);
+                return primitive{
+                    hpx::local_new<primitives::variable>(node_result)};
+            }
+        }
+        catch (const std::exception& ex)
+        {
+            PyErr_SetString(PyExc_RuntimeError, ex.what());
+        }
+        catch (...)
```

Comment:
You might want to consider catching `hpx:exception`'s as well.

Reply:
hpx::exception subclasses std::exception, so it seems unnecessary.

Reply:
`hpx::exception` might give you more error context information than what `std:;exception::what` returns. But sure - let's leave it as is for now.